src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -764,16 +765,27 @@ ExecuteSingleTask(QueryDesc *queryDesc, Task *task,
 		ShardPlacement *taskPlacement = (ShardPlacement *) lfirst(taskPlacementCell);
 		bool queryOK = false;
 		bool failOnError = false;
+		PGconn *connection = NULL;
 		int64 currentAffectedTupleCount = 0;
-		PGconn *connection = GetConnectionForPlacement(taskPlacement,
-													   isModificationQuery);
-
-		if (connection == NULL)
+		int connectionFlags = SESSION_LIFESPAN;
+		MultiConnection *multiConnection = GetNodeConnection(connectionFlags,
+															 taskPlacement->nodeName,
+															 taskPlacement->nodePort);
+		if (PQstatus(multiConnection->pgConn) != CONNECTION_OK)
 		{
 			failedPlacementList = lappend(failedPlacementList, taskPlacement);
+
```

Comment:
@anarazel Your comment on this line is very critical.

As far as I can see&test, it should be like this, right? The tests show me that this change conserves the existing behavior.

(I tested with single node failure, all nodes fails, a node fails while a transaction is going etc.)